src/analysis/processing/qgsalgorithmreclassifybylayer.cpp

```diff
@@ -0,0 +1,298 @@
+  request.setSubsetOfAttributes( QgsAttributeList() << mMinFieldIdx << mMaxFieldIdx << mValueFieldIdx );
+  mTableIterator = tableSource->getFeatures( request );
+
+  return true;
+}
+
+QVector<QgsReclassifyUtils::RasterClass> QgsReclassifyByLayerAlgorithm::createClasses( QgsRasterRange::BoundsType boundsType, const QVariantMap &, QgsProcessingContext &, QgsProcessingFeedback * )
+{
+  QVector< QgsReclassifyUtils::RasterClass > classes;
+  QgsFeature f;
+  while ( mTableIterator.nextFeature( f ) )
+  {
+    bool ok = false;
+    double minValue = f.attribute( mMinFieldIdx ).toDouble( &ok );
+    if ( !ok )
+      throw QgsProcessingException( QObject::tr( "Invalid value for minimum: %1" ).arg( f.attribute( mMinFieldIdx ).toString() ) );
+    double maxValue = f.attribute( mMaxFieldIdx ).toDouble( &ok );
+    if ( !ok )
+      throw QgsProcessingException( QObject::tr( "Invalid value for maximum: %1" ).arg( f.attribute( mMaxFieldIdx ).toString() ) );
+    double value = f.attribute( mValueFieldIdx ).toDouble( &ok );
+    if ( !ok )
+      throw QgsProcessingException( QObject::tr( "Invalid output value: %1" ).arg( f.attribute( mValueFieldIdx ).toString() ) );
+
+    classes << QgsReclassifyUtils::RasterClass( minValue, maxValue, boundsType, value );
+  }
+  return classes;
+}
+
+
+//
+// QgsReclassifyByTableAlgorithm
+//
+
+QString QgsReclassifyByTableAlgorithm::name() const
+{
+  return QStringLiteral( "reclassifybytable" );
+}
+
+QString QgsReclassifyByTableAlgorithm::displayName() const
+{
+  return QObject::tr( "Reclassify by table" );
+}
+
+QStringList QgsReclassifyByTableAlgorithm::tags() const
+{
+  return QObject::tr( "raster,reclassify,classes,calculator" ).split( ',' );
+}
+
+QString QgsReclassifyByTableAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm reclassifies a raster band by assigning new class values based on the ranges specified in a fixed table." );
+}
+
+QgsReclassifyByTableAlgorithm *QgsReclassifyByTableAlgorithm::createInstance() const
+{
+  return new QgsReclassifyByTableAlgorithm();
+}
+
+void QgsReclassifyByTableAlgorithm::addAlgorithmParams()
+{
+  addParameter( new QgsProcessingParameterMatrix( QStringLiteral( "TABLE" ),
+                QObject::tr( "Reclassification table" ),
+                1, false, QStringList() << QObject::tr( "Minimum" )
+                << QObject::tr( "Maximum" )
+                << QObject::tr( "Value" ) ) );
+}
+
+bool QgsReclassifyByTableAlgorithm::_prepareAlgorithm( const QVariantMap &, QgsProcessingContext &, QgsProcessingFeedback * )
+{
+  return true;
+}
+
+QVector<QgsReclassifyUtils::RasterClass> QgsReclassifyByTableAlgorithm::createClasses( QgsReclassifyUtils::RasterClass::BoundsType boundsType, const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  const QVariantList table = parameterAsMatrix( parameters, QStringLiteral( "TABLE" ), context );
+  int rows = table.count() / 3;
```

Comment:
To be fully homogeneous with the content of this method, I think that `rows` should be a `const` variable too :).